File: java/src/tree/RotateTest.java
package tree;

import org.junit.Test;

public class RotateTest {
	@Test
	public void testRandom() {
		for (var i = 0; i < 10_000; i++) {
			var td = new randomTestData(100);
			for (var j = 0; j < 100; j++) {
				Rotate.rotate(td.randomNode());
				td.assertNodes();
			}
		}
	}
}

File: go/tree/rotate_test.go
package tree

import "testing"

// TODO: fix broken test
func xTestRandomCases(t *testing.T) {
	for range 10_000 {
		td := newRandomTestData(100)
		for range 100 {
			rotate(td.randomNode(t))
			td.assertNodes(t)
		}
	}
}
